src/BCEngineMCMC.cxx

```diff
@@ -1384,6 +1384,19 @@ void BCEngineMCMC::Remarginalize(bool autorange)
     // combine statistics
     for (unsigned c = 0; c < fMCMCNChains; ++c)
         fMCMCStatistics_AllChains += fMCMCStatistics[c];
+
+}
+
+// --------------------------------------------------------
+void BCEngineMCMC::PrepareToContinueMarginalization(const std::string& filename, const std::string& mcmcTreeName, const std::string& parameterTreeName, bool loadObservables, bool autorange)
+{
+    LoadMCMC(filename, mcmcTreeName, parameterTreeName, loadObservables);
+    Remarginalize(autorange);
+    // delete trees
+    delete fMCMCTree;
```

Comment:
Do you mean why do I delete the trees? Because they are only used for loading in information. The trees that are held by `BCModel` are generally used for storing information; and we don't want to store more information into the file that was loaded in.